### ISSUE TITLE
Clearing SearchBox doesn't work properly when using a query with {searchTerms}

### ISSUE THREAD

**Issue by jhholm** [microsoft-search/pnp-modern-search#3197]
**Version used**
4.9.2

**Describe the bug**
Clearing the SearchBox doesn't actually clear the query text when using {searchTerms} in the Query Template with SharePoint Search. 

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Add a Search Box WebPart - leave at default configuration
2. Add a Search Results WebPart - with only the following changes
2.1. Use SharePoint Search as the data source
2.2. Change the query template to **{searchTerms} PromotedState:0**
2.2.1. This will ensure you should get results even with an empty queryText
2.4. Connect the Search Box WebPart via Available connections with the following configuration
![image](https://github.com/microsoft-search/pnp-modern-search/assets/13876124/fb848041-42d8-4711-8f6f-5f3a966aa344)

3. Publish the page
4. You should now get a results of all the content pages in your tenant
5. Perform e.g. a single word search in the Search Box that should give you less results but some
6. Note the Results count part of the Search Results WebPart shows less results and the query text
7. Clear the Search Box with the X button on the Search Box
8. The Results count doesn't show the query text anymore, but you are still getting the same results as before clearing the Search Box

**Expected behavior**
You should get the results as in step 4

**Ways to circumvent this**
If you change the query template to **{searchBoxQuery} PromotedState:0** this bug won't reproduce.

**Additional Context**
This seems to be a regression that came with #3028.  When you initially perform a search, TokenService is setting {searchTerms} with the text set in the Search Box. After clearing the Search Box the query still uses the old value in the querytemplate. My  first approach here would be to completely comment out the following lines:

https://github.com/microsoft-search/pnp-modern-search/blob/284e2bd8d1d8a1066d156e7224f5f10c106ca89c/search-parts/src/webparts/searchResults/SearchResultsWebPart.ts#L2246-L2249

I'm not sure if this is applicable when using another data source other than SharePoint though. SharePoint Search service will properly set the {searchTerms} under the hood if provided with a querytext. I quite often use {searchBoxQuery} instead of {searchTerms} as I noticed the same issue that #3028 was supposed to fix.

**Comment by kasperbolarsen** [comment]
Hi Janne, I can confirm your finding, however I *think* the intended way to achieve the request behavior is to:
Set the default value for the Search Results to the value you need (often *) AND set the property "Reset query in clear" to On in the Seach Box wp.

This approach makes sense to me as there is no way to know what the query should reset to, unless explicitly stated.

Will the above solve your issue?

**Comment by jhholm** [comment]
You are correct that using a default query * can be used to circumvent this issue. I am currently circumventing this issue with using {searchBoxQuery} instead of {searchTerms}.

You don't actually need a default value * if you have anything else in the query template than {searchTerms}. Microsoft search will handle a QueryTemplate with a {searchTerms} and an empty querytext:

In my example, when loading the page the initial search postquery sends the following values and receives the following QueryModification
```
QueryTemplate: "{searchTerms} PromotedState:0"
Querytext: ""

QueryModification: "PromotedState:0 -ContentClass=urn:content-class:SPSPeople"
```

After doing a search with the text "Test"
```
QueryTemplate: "Test PromotedState:0"
Querytext: "Test"

QueryModification: "Test PromotedState:0 -ContentClass=urn:content-class:SPSPeople"
```

After clearing the search box, you can see that the QueryTemplate still has the old value, and not sending {searchTerms} as on the initial load
```
QueryTemplate: "Test PromotedState:0"
Querytext: ""

QueryModification: "Test PromotedState:0 -ContentClass=urn:content-class:SPSPeople"
```

So I should get the same values sent as on the initial load, but the current implementation only checks if there is a value, not if the value has changed.

I think we should be able to let SharePoint Search handle populating {searchTerms} with the Querytext, but I am not completely sure if there are some cases where we need to have the token service populating the token. But as I mentioned I can use {searchBoxQuery} to have this working as the same way with commenting out the lines.

I can create a PR if needed. I currently can think of two options

1. Commenting out lines 2246-2249 as I think SharePoint Search can handle this as we are sending the querytext anyway
2. Changing the logic introduced by #3208 that it checks if the inputQueryText has changed from the last set value

**Comment by wobba** [comment]
@jhholm happy to get a PR to resolve this for {searchTerms} as well as I think we should be able to use that interchangably with {searchBoxQuery}.

**Comment by jhholm** [comment]
It seems so far that M365 Search 1.0 API does work with query templates somewhat similarly. Need to continue investigating though.

**Comment by wobba** [comment]
@jhholm the API support do support it but in the web part we replace {searchTerms} before passing the request I believe. resolving all we can clientside.

**Comment by jhholm** [comment]
I just applied the simplest fix for now, as I quickly noticed that a proper fix might take some extra thinking and refactoring.

**Comment by wobba** [comment]
Fixed with #3305
